### ISSUE TITLE
First step detection is not working when used multiple times in a composite actions

### ISSUE THREAD

**Issue by leonard84** [gradle/gradle-build-action#132]
If the `gradle-build-action` is used multiple times, only the first usage should configure the caching behavior, e.g. `cache-read-only`. This doesn't seem to work properly when used multiple times insight a composite action.

See https://github.com/spockframework/spock/runs/4651574620?check_suite_focus=true for example.

![image](https://user-images.githubusercontent.com/1151511/147639211-d612719a-be65-4777-b5b1-297aea648af2.png)

```yaml
name: Setup Gradle and JDKs
description: Sets up Gradle and JDKs

runs:
  using: "composite"
  steps:
    - name: Init Gradle Build Action
      uses: gradle/gradle-build-action@v2
      with:
        # cache options only count for the first invocation
        cache-read-only: ${{ github.ref != 'refs/heads/master' }}
        arguments: --version
    - name: 'Gradle javaToolchains'
      uses: gradle/gradle-build-action@v2
      with:
        arguments: javaToolchains
```

The second invocation will upload the cache entry.

**Comment by bigdaz** [comment]
Thanks for the report @leonard84 . Can you please test again with (just released) v2.1?
The "first step" detection is different in that version and I was not able to reproduce this issue. Thanks.

**Comment by leonard84** [comment]
The new version, seems to have fixed the problem https://github.com/spockframework/spock/runs/4689750253?check_suite_focus=true
